cli/slsa-verifier/main.go

```diff
@@ -61,17 +70,18 @@ func main() {
 		os.Exit(1)
 	}
 
-	verifiedProvenance, _, err := runVerify(artifactPath, provenancePath, source,
+	verifiedProvenance, _, err := runVerify(artifactReference, artifactPath, provenancePath, source,
```

Comment:
Maybe let's separate the functions to avoid special checks depending on options passed: `runVerifyArtifactPath()` and `runVerifyArtifactReference()`?

Trying to find a way to simplify the code, at the expense of a little bit of code duplication. Mixing two different logic is prone to mistakes - happened a lot in openssl in the past.

Reply:
Right now there's not difference in options, but there is a difference in artifact hash. Going to do this in the same follow-up PR as separating into subcommands. creating issue